ipn/ipnauth/ipnauth.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package ipnauth controls access to the LocalAPI.
+package ipnauth
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+	"os"
+	"os/user"
+	"runtime"
+	"strconv"
+	"syscall"
+
+	"inet.af/peercred"
+	"tailscale.com/net/netstat"
+	"tailscale.com/safesocket"
+	"tailscale.com/types/logger"
+	"tailscale.com/util/groupmember"
+	"tailscale.com/util/pidowner"
+	"tailscale.com/util/winutil"
+	"tailscale.com/version/distro"
+)
+
+// ConnIdentity represents the owner of a localhost TCP or unix socket connection
+// connecting to the LocalAPI.
+type ConnIdentity struct {
+	conn       net.Conn
+	notWindows bool // runtime.GOOS != "windows"
```

Comment:
I know that this is all code movement, but ... any reason that we're storing this? Now that everything is an accessor, we could replace `NotWindows()` with `return runtime.GOOS != "windows"` and make it easier for the compiler to optimize.

Reply:
I wish I remembered. I could make up some history but I'm sure the original is funnier.

Probably some half-done & forgotten migration? Maybe this time'll get it done! 🙃